testsuite/elytron/src/test/java/org/wildfly/test/integration/elytron/sasl/mgmt/AbstractMgmtSaslTestBase.java

```diff
@@ -0,0 +1,346 @@
+                .withRoleMapper(NAME).build());
+
+        elements.add(new ConfigurableElement() {
+
+            private void addUserWithDigestPass(CLIWrapper cli, String algorithm) {
+                cli.sendLine(
+                        String.format("/subsystem=elytron/filesystem-realm=%s:add-identity(identity=%s)", NAME, algorithm));
+                cli.sendLine(String.format(
+                        "/subsystem=elytron/filesystem-realm=%1$s:set-password(identity=%2$s, digest={algorithm=%2$s, password=%2$s%3$s, realm=%1$s})",
+                        NAME, algorithm, PASSWORD_SFX));
+            }
+
+            @Override
+            public void create(CLIWrapper cli) throws Exception {
+                cli.sendLine(String.format(
+                        "/subsystem=elytron/security-domain=ManagementDomain:write-attribute(name=trusted-security-domains, value=[%s])",
+                        NAME));
+
+                // identities with digested PWD
+                addUserWithDigestPass(cli, DIGEST_ALGORITHM_MD5);
+                addUserWithDigestPass(cli, DIGEST_ALGORITHM_SHA);
+                addUserWithDigestPass(cli, DIGEST_ALGORITHM_SHA256);
+                addUserWithDigestPass(cli, DIGEST_ALGORITHM_SHA512);
+            }
+
+            @Override
+            public void remove(CLIWrapper cli) throws Exception {
+                cli.sendLine(
+                        "/subsystem=elytron/security-domain=ManagementDomain:undefine-attribute(name=trusted-security-domains)");
+                // no need to remove identities, they'll be resolved with removing the FS realm
+            }
+
+            @Override
+            public String getName() {
+                return "domain-trust-and-identities";
+            }
+        });
+
+        elements.add(SimpleConfigurableSaslServerFactory.builder().withName(NAME).withSaslServerFactory("elytron")
+                .addFilter(SaslFilter.builder().withPatternFilter(saslMechanismName).build()).build());
+        elements.add(
+                SimpleSaslAuthenticationFactory.builder().withName(NAME).withSaslServerFactory(NAME).withSecurityDomain(NAME)
+                        .addMechanismConfiguration(MechanismConfiguration.builder().withMechanismName("OAUTHBEARER")
+                                .withRealmMapper("JWT").build())
+                        .addMechanismConfiguration(
+                                MechanismConfiguration.builder().withMechanismName(saslMechanismName)
+                                        .addMechanismRealmConfiguration(
+                                                MechanismRealmConfiguration.builder().withRealmName(NAME).build())
+                                        .build())
+                        .build());
+
+        elements.add(SimpleSocketBinding.builder().withName(NAME).withPort(PORT_NATIVE).build());
+        elements.add(SimpleMgmtNativeInterface.builder().withSocketBinding(NAME).withSaslAuthenticationFactory(NAME).build());
```

Comment:
IMO, the class names are self-descriptive enough:
* SimpleSocketBinding - `/socket-binding-group=standard-sockets/socket-binding=[NAME]`
* SimpleMgmtNativeInterface - `/core-service=management/management-interface=native-interface`